### PR TITLE
Make all Meteor.Error calls consistent

### LINKED ISSUE #3122: Make all `Meteor.Error` calls consistent
This is just a "cleanup" task

`Meteor.Error` should be `Meteor.Error("short-message", "Long message with more detail");`

Currently we have a hodgepodge of different styles. Specifically we want to eliminate versions that use a number instead of a message as these don't really convey any helpful info.

### PR DESCRIPTION
**Pull request by codejockie**
Resolves #3122 

## How To Test
In an editor, search for all instances of `Meteor.Error` check that it conforms with this style `Meteor.Error("short-message", "Long message with more detail")`

### CODE REVIEW

**Comment by zenweasel** [comment]
I feel like trying to make these messages fit within the confines of HTTP response code is a leaky abstraction. Quickly you get lots of things that don't fit in that mold or sort of do but the HTTP message is not _quite_ what you want. I think the main thing is that the short messages need to be consistent. So every access denied needs to be `access-denied` not sometimes `forbidden`.

For me keeping the short messages consistent is about making the log files easily grep-able.

File: server/methods/accounts/accounts.js [hunk 1]
```diff
@@ -667,11 +667,11 @@ export function inviteShopMember(options) {
 
   // check to ensure that invitee has roles required to perform the invitation
   if (!Reaction.canInviteToGroup({ group, user: Meteor.user() })) {
-    throw new Meteor.Error(403, "cannot invite to group");
+    throw new Meteor.Error("access-denied", "cannot invite to group");
   }
 
   if (group.slug === "owner") {
-    throw new Meteor.Error(400, "cannot directly invite owner");
+    throw new Meteor.Error("bad-request", "cannot directly invite owner");
```
Comment (zenweasel) [inline server/methods/accounts/accounts.js]: Should start with capital letter

**Comment by zenweasel** [comment]
I think some of these that are `invalid-thing` or `cannot-find-thing` could be switched to `invalid-parameter` with the long message being more descriptive. All the `access-denied`'s that are server-side can be made more explicit stating what access what denied, but leave the client-side ones vague.  ("You do not have access to modify products")

That's my opinion at least. Open to more suggestions and hopefully something that we can codify into a style rule.

File: server/methods/catalog.js [hunk 2]
```diff
@@ -1394,12 +1394,12 @@ Meteor.methods({
         });
       } else {
         Logger.debug("invalid product visibility ", productId);
-        throw new Meteor.Error("Forbidden", "Variant is required");
+        throw new Meteor.Error("forbidden", "Variant is required");
       }
```
Comment (zenweasel) [inline server/methods/catalog.js]: this isn't really forbidden. It seems like invalid parameters.

File: server/methods/catalog.js [hunk 3]
```diff
@@ -1421,7 +1421,7 @@ Meteor.methods({
       return res === 1 && !product.isVisible;
     }
     Logger.debug("invalid product visibility ", productId);
-    throw new Meteor.Error("Bad Request");
+    throw new Meteor.Error("bad-request", "Bad Request");
```
Comment (zenweasel) [inline server/methods/catalog.js]: should be `invalid-parameters`

File: server/methods/core/cart.js [hunk 4]
```diff
@@ -348,7 +348,7 @@ Meteor.methods({
     const cart = Collections.Cart.findOne({ userId: this.userId });
     if (!cart) {
       Logger.error(`Cart not found for user: ${ this.userId }`);
-      throw new Meteor.Error(404, "Cart not found",
+      throw new Meteor.Error("cart-not-found",
```
Comment (zenweasel) [inline server/methods/core/cart.js]: Probably should be `invalid-parameter`

File: server/methods/core/cart.js [hunk 5]
```diff
@@ -374,12 +374,12 @@ Meteor.methods({
     // const variant = Collections.Products.findOne(variantId);
     if (!product) {
       Logger.warn(`Product: ${ productId } was not found in database`);
-      throw new Meteor.Error(404, "Product not found",
+      throw new Meteor.Error("product-not-found",
         "Product with such id was not found!");
```
Comment (zenweasel) [inline server/methods/core/cart.js]: Let's remove exclamation points. Stay calm everyone.

File: server/methods/core/cart.js [hunk 6]
```diff
@@ -845,14 +845,14 @@ Meteor.methods({
...
-        throw new Meteor.Error("An error occurred adding the address");
+        throw new Meteor.Error("server-error", "An error occurred adding the address");
       }
     }
     // refresh shipping quotes
     Meteor.call("shipping/updateShipmentQuotes", cartId);
 
     if (typeof cart.workflow !== "object") {
-      throw new Meteor.Error(500, "Internal Server Error",
+      throw new Meteor.Error("internal-server-error",
```
Comment (zenweasel) [inline server/methods/core/cart.js]: This should be just `server-error` like the rest.

File: server/methods/core/groups.js [hunk 7]
```diff
@@ -55,13 +55,13 @@ Meteor.methods({
...
     if (groupExists) {
-      throw new Meteor.Error(409, "Group already exist for this shop");
+      throw new Meteor.Error("conflict", "Group already exist for this shop");
     }
     try {
       _id = Groups.insert(newGroupData);
     } catch (error) {
       Logger.error(error);
-      throw new Meteor.Error(400, "Bad request");
+      throw new Meteor.Error("bad-request", "Bad request");
```
Comment (zenweasel) [inline server/methods/core/groups.js]: Let's make this `invalid-parameter`

File: server/methods/core/groups.js [hunk 8]
```diff
@@ -100,7 +100,7 @@ Meteor.methods({
 
     // prevent edits on owner. Owner groups is the default containing all roles, and as such should be untouched
     if (group.slug === "owner") {
-      throw new Meteor.Error(400, "Invalid request");
+      throw new Meteor.Error("bad-request", "Invalid request");
```
Comment (zenweasel) [inline server/methods/core/groups.js]: Let's make this `invalid-parameter` and the long explanation a little more helpful if possible.

File: server/methods/core/groups.js [hunk 9]
```diff
@@ -216,11 +216,11 @@ Meteor.methods({
     // we are limiting group method actions to only users with admin roles
     // this also include shop owners, since they have the `admin` role in their Roles.GLOBAL_GROUP
     if (!Reaction.hasPermission("admin", Meteor.userId(), shopId)) {
-      throw new Meteor.Error(403, "Access Denied");
+      throw new Meteor.Error("access-denied", "Access Denied");
     }
 
     if (!user) {
-      throw new Meteor.Error(404, "Could not find user");
+      throw new Meteor.Error("not-found", "Could not find user");
```
Comment (zenweasel) [inline server/methods/core/groups.js]: invalid-parameter

**Comment by spencern** [comment]
@zenweasel @codejockie 
My thinking on this seems to be inline with Brent here. I'd like to stay consistent with error messages where it makes sense, but agree that we shouldn't be dogmatic about sticking to HTTP status code style messages when it doesn't fit.

What I'd like to do is provide some type of style-guide for these messages so that people writing code don't have to stop to figure out what their error message should say, it can be open-ended in the sense that we can always add a new `short-message` as the need arises, but we can attempt to keep things consistent when possible.

@machikoyasuda let's figure out if there are some patterns we can adopt from this PR that we can use to start a style guide for this (and perhaps look around to see if any other open-source repos have good examples)

**Comment by zenweasel** [comment]
@spencern @machikoyasuda Should we just approve/merge this PR for now and revisit when we have a standard? I feel like this is definitely a big improvement over the existing implementation

**Pull request merged**